api/cds.proto

```diff
@@ -174,56 +174,59 @@ message Cluster {
   repeated Address dns_resolvers = 18;
 
   // If specified, outlier detection will be enabled for this upstream cluster.
+  // See the :ref:`architecture overview <arch_overview_outlier_detection>` for more information on outlier
+  // detection. Each of the configuration values can be overridden via
+  // :ref:`runtime values <config_cluster_manager_cluster_runtime_outlier_detection>`.
   message OutlierDetection {
     // The number of consecutive 5xx responses before a consecutive 5xx ejection
     // occurs. Defaults to 5.
-    google.protobuf.UInt32Value consecutive_5xx = 1;
+    google.protobuf.UInt32Value consecutive_5xx = 1 [(validate.rules).uint32.gt = 0];
     // The time interval between ejection analysis sweeps. This can result in
     // both new ejections as well as hosts being returned to service. Defaults
     // to 10000ms or 10s.
-    google.protobuf.Duration interval = 2;
+    google.protobuf.Duration interval = 2 [(validate.rules).duration.gt.nanos = 0];
     // The base time that a host is ejected for. The real time is equal to the
     // base time multiplied by the number of times the host has been ejected.
     // Defaults to 30000ms or 30s.
-    google.protobuf.Duration base_ejection_time = 3;
+    google.protobuf.Duration base_ejection_time = 3 [(validate.rules).duration.gt.nanos = 0];
     // The maximum % of an upstream cluster that can be ejected due to outlier
     // detection. Defaults to 10%.
-    google.protobuf.UInt32Value max_ejection_percent = 4;
+    google.protobuf.UInt32Value max_ejection_percent = 4 [(validate.rules).uint32 = {gte:0, lte:100}];
```

Comment:
nit: gte 0 is redundant as this is uint. same elsewhere.

Reply:
I was debating this one, because I didn't know if we wanted to "catch" that at validate level or the protobuf field level. I have to admit I didn't try to see the failure, but in spirits of eliminating redundancy I see your point. Will update.